### ISSUE TITLE
Default code flagged

### ISSUE THREAD

**Issue by igroti** [realm/SwiftLint#5799]
Swiflint: 0.57.0
```
guard scene as? UIWindowScene != nil else { return }
```
This code comes with Xcode by default and is getting flagged. I don't understand what the error is expecting:
```
Prefer Type Checking Violation: Prefer `a is X` to `a as? X != nil` (prefer_type_checking)
```

**Comment by SimplyDanny** [comment]
The rule suggests to rewrite this as
```swift
guard scene is UIWindowScene else { return }
```
